runtime/doc/eval.txt

```diff
@@ -2664,6 +2666,36 @@ delete({fname})							*delete()*
 		To delete a line from the buffer use |:delete|.  Use |:exe|
 		when the line number is in a variable.
 
+dictwatcheradd({dict}, {pattern}, {callback})		      *dictwatcheradd()*
+		Adds a watcher to a dictionary. A dictionary watcher is
+		identified by three components:
+
+		- A dictionary({dict});
+		- A key pattern({pattern}).
+		- A function({callback}).
+
+		After this is called, every change on {dict} and on keys
+		matching {pattern} will result in {callback} being invoked.
+		The {callback} receives two arguments:
+
+		- The key which changed.
+		- A dictionary containg the new and old values for the key.
```

Comment:
I would also highly suggest to pass also a dictionary being watched. And I think you may simplify the code thus, because in this case only needed things are

1. Dictionary being watched.
2. Changed key.
3. (optional) Old value. (Or new value, I did not check when you call this watcher: before or after value changed.)

I.e. function signature will be `watcher(dict, key, ...)`. This saves you up to four allocates (dictionary, dictionary item for old value, dictionary item for new value, hash table allocate). Passing dictionary for which notification was created allows using notifications not only for options: e.g. I can imagine how you can use this feature to implement monkey-patching in some OOP implementations (i.e. you have a class dictionary where you put new method, but since instance dictionaries are separate to attach new method to all instances you need to patch them. Patching may be done in notification function, but it would be very, *very* inconvenient to implement without passing `dict` argument).